src/util/MetaUtils.cpp

```diff
@@ -78,6 +78,57 @@ normalizeOps(xdr::xvector<OperationMeta>& oms)
 namespace stellar
 {
 
+void
+normalizeMeta(LedgerCloseMeta& m)
+{
+    switch (m.v())
+    {
+    case 0:
+        for (auto& trm : m.v0().txProcessing)
+        {
+            normalizeMeta(trm);
+        }
+        for (auto& up : m.v0().upgradesProcessing)
+        {
+            sortChanges(up.changes);
+        }
+        break;
+    case 1:
+        for (auto& trm : m.v1().txProcessing)
+        {
+            normalizeMeta(trm);
+        }
+        for (auto& up : m.v1().upgradesProcessing)
+        {
+            sortChanges(up.changes);
+        }
+        break;
+    case 2:
+        for (auto& trm : m.v2().txProcessing)
+        {
+            normalizeMeta(trm);
+        }
+        for (auto& up : m.v2().upgradesProcessing)
+        {
+            sortChanges(up.changes);
+        }
+        std::sort(m.v2().evictedPersistentLedgerEntries.begin(),
+                  m.v2().evictedPersistentLedgerEntries.end());
+        std::sort(m.v2().evictedTemporaryLedgerKeys.begin(),
+                  m.v2().evictedTemporaryLedgerKeys.end());
+        break;
+    default:
+        releaseAssert(false);
+    }
+}
+
+void
+normalizeMeta(TransactionResultMeta& m)
+{
+    sortChanges(m.feeProcessing);
+    normalizeMeta(m.txApplyProcessing);
+}
+
 void
 normalizeMeta(TransactionMeta& m)
```

Comment:
I ran this normalization algorithm on a few ledgers and I found one case where it produced unexpected output. If you look at the transaction result meta for tx [ac4a2c880fe74ddbbe2d6b3ab8d14eb6b10db7abe5d3359fc853161fb0fcf7a6](https://horizon.stellar.org/transactions/ac4a2c880fe74ddbbe2d6b3ab8d14eb6b10db7abe5d3359fc853161fb0fcf7a6), you will find the following changes in `txChangesBefore`:

```
txChangesBefore: Array[4]
[0]: [ledgerEntryState]
state
lastModifiedLedgerSeq: 48249545
data: [account]
account
accountId: [publicKeyTypeEd25519]
ed25519: GB2SZS3NO5NR2JQNVDZ5VEFPSXYL6CT3ZZ3JEBEDONN6GZZZYGTNH7VX
balance: 96676506
seqNum: 204162468711237283
numSubEntries: 0
inflationDest: none
flags: 0
homeDomain:
thresholds: AQAAAA==
signers: Array[0]
ext: [undefined]
v1
liabilities
buying: 0
selling: 0
ext: [undefined]
v2
numSponsored: 0
numSponsoring: 0
signerSponsoringIDs: Array[0]
ext: [undefined]
v3
ext: [undefined]
seqLedger: 48249077
seqTime: 1695514068
ext: [undefined]

[1]: [ledgerEntryUpdated]
updated
lastModifiedLedgerSeq: 48249545
data: [account]
account
accountId: [publicKeyTypeEd25519]
ed25519: GB2SZS3NO5NR2JQNVDZ5VEFPSXYL6CT3ZZ3JEBEDONN6GZZZYGTNH7VX
balance: 96676506
seqNum: 204162468711237283
numSubEntries: 0
inflationDest: none
flags: 0
homeDomain:
thresholds: AQAAAA==
signers: Array[0]
ext: [undefined]
v1
liabilities
buying: 0
selling: 0
ext: [undefined]
v2
numSponsored: 0
numSponsoring: 0
signerSponsoringIDs: Array[0]
ext: [undefined]
v3
ext: [undefined]
seqLedger: 48249077
seqTime: 1695514068
ext: [undefined]

[2]: [ledgerEntryState]
state
lastModifiedLedgerSeq: 48249545
data: [account]
account
accountId: [publicKeyTypeEd25519]
ed25519: GB2SZS3NO5NR2JQNVDZ5VEFPSXYL6CT3ZZ3JEBEDONN6GZZZYGTNH7VX
balance: 96676506
seqNum: 204162468711237283
numSubEntries: 0
inflationDest: none
flags: 0
homeDomain:
thresholds: AQAAAA==
signers: Array[0]
ext: [undefined]
v1
liabilities
buying: 0
selling: 0
ext: [undefined]
v2
numSponsored: 0
numSponsoring: 0
signerSponsoringIDs: Array[0]
ext: [undefined]
v3
ext: [undefined]
seqLedger: 48249077
seqTime: 1695514068
ext: [undefined]

[3]: [ledgerEntryUpdated]
updated
lastModifiedLedgerSeq: 48249545
data: [account]
account
accountId: [publicKeyTypeEd25519]
ed25519: GB2SZS3NO5NR2JQNVDZ5VEFPSXYL6CT3ZZ3JEBEDONN6GZZZYGTNH7VX
balance: 96676506
seqNum: 204162468711237284
numSubEntries: 0
inflationDest: none
flags: 0
homeDomain:
thresholds: AQAAAA==
signers: Array[0]
ext: [undefined]
v1
liabilities
buying: 0
selling: 0
ext: [undefined]
v2
numSponsored: 0
numSponsoring: 0
signerSponsoringIDs: Array[0]
ext: [undefined]
v3
ext: [undefined]
seqLedger: 48249545
seqTime: 1695516758
ext: [undefined]
```

After sorting these changes it will rearrange them so that the `ledgerEntryState` items are adjacent to each other. That is because all the ledger entry changes have the same ledger key (so the tie breaker in the comparison is the ledger entry type).